shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -466,6 +533,214 @@ fun NearbyStaticData.withRealtimeInfo(
         .sortedWith(compareBy(Route.relevanceComparator(pinnedRoutes)) { it.sortRoute() })
 }
 
+/**
+ * For reasons, for mid-line subway shuttles/suspensions, schedules will correctly reflect the
+ * closure and use route patterns that stop at temporary terminals, but predictions will use the
+ * canonical route patterns. The way we work around this is that for
+ * 1. subway routes
+ * 2. with alerts somewhere on the line
+ * 3. where the schedule has non-typical patterns instead of typical patterns
+ * 4. but the predictions all have typical patterns instead of non-typical patterns
+ *
+ * we check the stops of the scheduled patterns against both the predicted pattern and the
+ * prediction's current stop to override the prediction's route pattern with a correct scheduled
+ * pattern, and then we load the headsign from the new route pattern's representative trip. This
+ * should be narrowly scoped enough to apply to temporary terminals but no other trips. Hopefully.
```

Comment:
praise: I appreciate the detailed comment here! Is there any relevant code from screenplay that is relevant to link here?

Reply:
Not that I know of. Why would there be? Am I missing something?

Reply:
It's [not](https://mbta.slack.com/archives/C03K6NLKKD1/p1726160321566569?thread_ts=1726096049.324989&cid=C03K6NLKKD1) - they're scanning the trip's full predictions to get its actual last stop and using that as the headsign, but we don't have the trip's full predictions, and I don't think we want them just for this.